src/Build/Graph/ProjectGraph.cs

```diff
@@ -159,14 +199,21 @@ public ProjectGraph(string entryProjectFile)
         }
 
         private ProjectGraphNode CreateNewNode(
-            string projectFilePath,
-            ProjectCollection projectCollection,
-            Dictionary<string, string> globalProperties,
-            string toolsVersion)
+            ConfigurationMetadata configurationMetadata,
+            ProjectCollection projectCollection)
         {
-            var project = new ProjectInstance(projectFilePath, globalProperties, toolsVersion, projectCollection);
-            var graphNode = new ProjectGraphNode(project);
-            _allParsedProjects.Add(projectFilePath, graphNode);
+            // TODO: ProjectInstance just converts the dictionary back to a PropertyDictionary, so find a way to directly provide it.
```

Comment:
Mainly just a perf TODO. IMHO, not something that needs to be addressed in this PR. Likely the solution is an internal ctor that only this code path uses, which is kinda meh.

Reply:
Yeah, I think this can wait.